tests/test_mcp_server.py

```
@@ -0,0 +1,425 @@
+# ── Agent loop MCP integration ────────────────────────────────────────────────
+
+class TestAgentLoopMCPIntegration:
+    """Ensure _dispatch_tool routes MCP-only tools correctly."""
+
+    def _make_runner(self, mcp_client=None):
+        from agent.loop import AgentRunner
+        runner = AgentRunner.__new__(AgentRunner)
+        runner._mcp = mcp_client
+        from agent.tools import WorkspaceTools
+        runner.tools = MagicMock(spec=WorkspaceTools)
+        runner.tools.root = Path("/tmp/fake-workspace")
```

Ruff (0.15.12)

[error] Probable insecure usage of temporary file or directory: "/tmp/fake-workspace"

(S108)